service/worker/replicator/replicationTask.go

```diff
@@ -515,6 +528,11 @@ func (t *workflowReplicationTask) Nack() {
 	t.metricsClient.RecordTimer(t.metricsScope, metrics.ReplicatorLatency, t.timeSource.Now().Sub(t.startTime))
 
 	t.state = task.TaskStateNacked
+
+	// TODO: need to discuss whether it is the good place to place the counter increment for the metrics because
```

Comment:
We can remove this TODO as we handle the retry ourselves and NAck means put it into DLQ